### PR TITLE
chore: Extract store factory

### LINKED ISSUE #13941: Extract store factory

### PR DESCRIPTION
**Pull request by netopyr**
**Description**:

This PR extracts the store factory from HandleContext

**Related issue(s)**:

Fixes #13941 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by lukelee-sl — DISMISSED** [review]
Smart contract changes LGTM.

**Review by IvanKavaldzhiev — DISMISSED** [review]
LGTM

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/child/modules/ChildDispatchModule.java [hunk 1]
```diff
@@ -130,6 +131,15 @@ static ServiceApiFactory provideServiceApiFactory(
         return new ServiceApiFactory(stack, configuration, storeMetricsService);
     }
 
+    @Provides
+    @ChildDispatchScope
+    static StoreFactoryImpl storeFactory(
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/child/modules/ChildDispatchModule.java:136]: We could probably return `StoreFactory` here.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/user/modules/UserDispatchModule.java [hunk 2]
```diff
@@ -123,6 +125,15 @@ static ServiceApiFactory provideServiceApiFactory(
         return new ServiceApiFactory(stack, configuration, storeMetricsService);
     }
 
+    @Provides
+    @UserDispatchScope
+    static StoreFactoryImpl storeFactoryImpl(
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/user/modules/UserDispatchModule.java:130]: same here

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM! Just had few minor comments. Thanks @netopyr

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/child/modules/ChildDispatchModule.java (hunk 1 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/child/modules/ChildDispatchModule.java]: Fixed, thanks @Neeharika-Sompalli !

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/user/modules/UserDispatchModule.java (hunk 2 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/user/modules/UserDispatchModule.java]: Fixed

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! Thanks @netopyr

**Review by Nana-EC — APPROVED** [review]

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/store/StoreFactory.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+     * that doesn't have a corresponding HAPI {@link TransactionBody}.
+     *
+     * @param apiInterface The API interface to find and create an implementation of
+     * @param <T> Interface class for an API
+     * @return An implementation of the provided API interface
+     * @throws IllegalArgumentException if the apiInterface class provided is unknown to the app
+     * @throws NullPointerException if {@code apiInterface} is {@code null}
+     */
+    @NonNull
+    <T> T serviceApi(@NonNull Class<T> apiInterface);
```
Comment (rbair23) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/store/StoreFactory.java:64]: Based on the name `StoreFactory` and the package `store`, I'm surprised to see `serviceApi` in this interface. Maybe this isn't just a `StoreFactory` and maybe it doesn't belong exclusively to the `store`?

Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/store/StoreFactory.java:64]: I agree that the name `StoreFactory` does not capture the intent. I have not found a better name yet. It probably has to mature a bit. 🙂

**Pull request merged**